### PR TITLE
General fixes

### PR DESCRIPTION
**Pull request by ThatOneCalculator**
- Puts configuration in standard directory
- Since it's designed to be run with bun, use `node:*` imports for builtins
- Format and lint with Biome

### CODE REVIEW

**Comment by ThatOneCalculator** [comment]
@kynnyhsap updated with your latest changes

File: src/systemPrompt.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import os from "os";
+import os from "node:os";
```
Comment (kynnyhsap) [inline src/systemPrompt.ts:1]: bun aims to have full compatibility with node, so I don't see the need in the prefix

File: package.json [hunk 2]
```diff
@@ -1,23 +1,25 @@
...
+	"scripts": {
+		"how": "bun run ./src/index.ts --",
+		"compile": "bun build ./src/index.ts --compile --minify --sourcemap --outfile how",
+		"check": "biome check ./src --fix"
+	},
+	"dependencies": {
+		"@inquirer/prompts": "^5.1.2",
+		"chalk": "^5.3.0",
+		"clipboardy": "^4.0.0",
+		"env-paths": "^3.0.0"
```
Comment (kynnyhsap) [inline package.json:16]: bun can read `.env` files out of the box. 

and i only needed `.env` to store my testing keys and that's why i added it in gitignore, they are not bundled with executable, they only for development.

File: package.json [hunk 3]
```diff
@@ -1,23 +1,25 @@
...
+	},
+	"devDependencies": {
+		"@types/bun": "latest",
+		"biome": "^0.3.3"
```
Comment (kynnyhsap) [inline package.json:20]: i use zed editor so i opted in by default to prettier, i will check out biome and see if i like it. seems nice

**Review by kynnyhsap — CHANGES_REQUESTED** [review]
thank you for the PR!

there has been a lot of changes since you've made the PR. sorry for not noticing it. i am not used to getting PR on my own repos =))

i left some comments.

btw installation script is ready and you can try it out for macos and linux without building from source

File: src/systemPrompt.ts (hunk 1 above)
Comment (ThatOneCalculator) [reply src/systemPrompt.ts:1]: I'm super confused why you think that having compatibility with node APIs would imply not using proper namespaces...

File: package.json (hunk 2 above)
Comment (ThatOneCalculator) [reply package.json:16]: You seem to be uninformed on what that package is for – it's for getting standard configuration directories (i.e. `$HOME/.config/how`)

File: package.json (hunk 3 above)
Comment (ThatOneCalculator) [reply package.json:20]: There's a biome plugin for zed :)

File: src/systemPrompt.ts (hunk 1 above)
Comment (kynnyhsap) [reply src/systemPrompt.ts:1]: https://x.com/jarredsumner/status/1811521754109800609?s=46

File: package.json (hunk 2 above)
Comment (kynnyhsap) [reply package.json:16]: i don't see why we need that package then.
i want to store config inside `$HOME/.how/config.json` path. i don't need external package for that.

File: package.json (hunk 3 above)
Comment (kynnyhsap) [reply package.json:20]: ok i tried biome. 
* didn't see much performance gain compared do prettier formatting (so this is not a killer feature for me)
* i like that compared to eslint it is only one package, but
* i don't like may of the rules and i don't want to go and disable them manually
 
for such a small projects i don't see a need in linters, they are more of a distraction

File: package.json (hunk 2 above)
Comment (ThatOneCalculator) [reply package.json:16]: First, applications should always use a standard directory. Storing your configuration in the home directory in its own dotfolder is considered very bad practice. Second, you are not considering people with filesystems that don't follow modern Unix like paths.

File: src/systemPrompt.ts (hunk 1 above)
Comment (ThatOneCalculator) [reply src/systemPrompt.ts:1]: Yes, they're optional. It's still ***best practice***.

**Comment by ThatOneCalculator** [comment]
Based on your unwillingness to change and lack of understanding of why best practices should be used even in small projects, I will be closing this PR. I tried to help you but it seems you're not receptive to change, I hope you have a good day.

File: package.json (hunk 2 above)
Comment (kynnyhsap) [reply package.json:16]: if it's in home it's easy to find and delete. a lot of tools locate themselves in home and i find it very nice.
Screenshot 2024-07-12 at 17 27 37

i don't plan to support non-unix systems. i made this tool for me and my friends. you are free to fork and make any changes you want. 

you should question your premise on "best practices"

10 years ago jquery and prototype inheritance was considered a "best practice"

Comment (ThatOneCalculator) [reply package.json:16]: There's a reason XDG standard directories exist. It makes your system far easier to back up and makes application configuration far simpler.

**Comment by ThatOneCalculator** [comment]
https://x.com/pashynnyk/status/1811772250091381104

If you would like to say something about me, say it to my face and not behind my back. 

I took time out of my day to fix issues with your code and all you did was complain and boil it down to "my way is better, it's a small project and it doesn't matter". I'm not saying that you have to like or even use my changes, but you were so quick to completely disregard what I had to say under the guise of "my way or the highway".

I hope you can take time to reflect upon how you treat others.

**Comment by kynnyhsap** [comment]
i'd love to say it in your face - you have an attitude. i thank you for your time on trying to make this project better, but i don't owe you anything

you did not convince be that these changes are necessary, sorry. i am open to changes sir, if they bring value. just adding other formatter does not bring any value.

i never said my "way is better". i didn't not say "it's a small project and it does not matter". on contrary, i deliberately kept the configuration and number of packages to a minimum. and also my plan is to rewrite in zig someday, so more js tooling is irrelevant. a few packages i used i used to quickly prototype. i will get rid of them eventually.

**Pull request closed without merging**